### ISSUE TITLE
Surround With

### ISSUE THREAD

**Issue by merqlove** [KronicDeth/intellij-elixir#720]
Sometimes, we need to wrap data with quotes or brackets.

## Action
Wrap with `[]`, `""`, `''`, by selecting DATA, then touching one of: `[`, `"`, `'`

### Initial
`DATA`

### BAD (currently)
`[]`

### GOOD
`[DATA]`

**Comment by KronicDeth** [comment]
I'm not sure about tying it to the container promoters (`[`, `"`, `'`), but in general IntelliJ calls this "Surround with..." and it's triggered by `Code > Surround With` or  `⌥⌘T`:
* https://blog.jetbrains.com/idea/2006/07/surround-with/
* https://www.jetbrains.com/help/idea/surrounding-blocks-of-code-with-language-constructs.html

There's probably an ExtensionPoint to add ones with plugins, that we can use to add this for Elixir.

You shouldn't call this `Wrap` because `Wrap` is the term used for wrapping lines that exceed pass the line length limit.

**Comment by merqlove** [comment]
Yeah, seems it work through ⌥⌘T under IDEA. Thanks for the notice. I'm asked this one, because It helps a lot within RM.

**Comment by KronicDeth** [comment]
I think it still needs to be customized for Elixir, when I tried it, it only prompted for `()` and `{}`, so I think it's using the Java templates by default.

**Comment by merqlove** [comment]
Yeah, sometimes i'm have angry with this behavior

**Comment by sh41** [comment]
This already works, and has for a long time — it just isn't plugin behaviour, which is why it wasn't obvious.

Selecting text and typing `[`, `(`, `{`, `<`, `"`, `'` or `` ` `` surrounds the selection with the matching pair. The IDE handles it for every language, and it is on by default: **Settings > Editor > General > Smart Keys > "Surround selection on typing quote or brace"**.

So `DATA` + `[` gives `[DATA]` in Elixir files today, with no plugin support needed.

`Code > Surround With` is a separate feature and still has no Elixir-specific entries — worth its own issue if anyone wants it.

Closed as part of an AI-assisted review of the backlog. A human checked this one before closing it, but it might still be a mistake. If you think the reason above is wrong, please reopen the issue and tell us why.
